rclcpp/include/rclcpp/topic_statistics/subscription_topic_statistics.hpp

```diff
@@ -152,6 +152,12 @@ class SubscriptionTopicStatistics
       publisher_->publish(msg);
     }
     window_start_ = window_end;
+    {
+      std::lock_guard<std::mutex> lock(mutex_);
+      for (auto & collector : subscriber_statistics_collectors_) {
+        collector->ClearCurrentMeasurements();
```

Comment:
What about calling this right after `statistics_msgs::msg::MetricsMessage` is created for each collector? i think it would be better for each collector to have less time window to reset.

Reply:
This is a good point.

I'm just trying to think how to do it while satisfying https://github.com/ros2/rclcpp/pull/1535#issuecomment-770429078. Maybe wrap the `publish()` in an async call or rename this member function to `publish_message_and_reset_measurements()`

Reply:
True.
The ClearCurrentMeasurements() is likely to be after GetStatisticsResults() at the earliest.